src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -114,7 +114,16 @@ public static bool TcpConnectionStringDoesNotUseAadAuth
             get
             {
                 SqlConnectionStringBuilder builder = new (TCPConnectionString);
-                return builder.Authentication == SqlAuthenticationMethod.SqlPassword || builder.Authentication == SqlAuthenticationMethod.NotSpecified;
+                return builder.Authentication == SqlAuthenticationMethod.SqlPassword || (builder.Authentication == SqlAuthenticationMethod.NotSpecified && AuthenticatingWithoutAccessToken);
+            }
+        }
+
+        // Testing in macOS Azure SQL container requires access token authentication
+        public static bool AuthenticatingWithoutAccessToken
+        {
+            get
+            {
+                return !RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```

Comment:
This more sense with positive naming. Feel free to modify the naming as you like.
```suggestion
        public static bool ForceAuthenticationWithAccessToken
        {
            get
            {
                return RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```